src/java/fr/paris/lutece/plugins/suggest/business/SuggestSubmitType.java

```diff
@@ -185,12 +185,7 @@ public int getIdXSLStyleSheet( )
      */
     public boolean equals( Object obj )
     {
-        if ( obj instanceof SuggestSubmitType && ( (SuggestSubmitType) obj ).getIdType( ) == _nIdType )
-        {
-            return true;
-        }
-
-        return false;
+        return obj != null && obj == this.getClass() && ( (SuggestSubmitType) obj ).getIdType( ) == _nIdType;
```

Comment:
this seems wrong. `obj` is almost certainly not the class `SuggestSubmitType`

Reply:
Ok, I'll rollback to instance of...